### PR TITLE
Compare cars redux part added

### PR DESCRIPTION
**Pull request by pant-manas**
Added the Compare car redux part
As discussed added two reducers for the same

Also looking into the strict mode part. Removed it for now for the other reviews

### CODE REVIEW

File: src/components/CarCard/CarCard.tsx [hunk 1]
```diff
@@ -1,25 +1,46 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import "./CarCard.scss";
 import rightarrow from "../../assets/rightArrow.svg";
-import { IcarData } from "../../models/ICarData";
+import { CarDetails } from "../../models/CarDetails";
+import { useDispatch, useSelector } from "react-redux";
+import { addToCompare } from "../../redux/CompareCars/reducer";
+import { NUMBEROFCARSTOCOMPARE } from "../../constants";
```
Comment (pashedin) [inline src/components/CarCard/CarCard.tsx]: you can use snake_case for cinstants if all caps is involved

File: src/redux/store.ts [hunk 2]
```diff
@@ -15,6 +18,8 @@ export const store = configureStore({
   reducer: {
     cart: CartReducer,
     authenticationReducer: authenticationreducer,
+    CompareCarReducer: comparecarreducer,
+    CompareCarsDetailsSlice: comparecardetailslice,
```
Comment (pashedin) [inline src/redux/store.ts]: use appropriate cases

**Review by pashedin — CHANGES_REQUESTED** [review]
just naming convention stuff

File: src/components/CarCard/CarCard.tsx (hunk 1 above)
Comment (pant-manas) [reply src/components/CarCard/CarCard.tsx]: @pashedin sure added the changes

File: src/redux/store.ts (hunk 2 above)
Comment (pant-manas) [reply src/redux/store.ts]: @pashedin sure added the changes in the files

**Review by pashedin — APPROVED** [review]

**Pull request merged**